addon/scheduler.js

```diff
@@ -0,0 +1,124 @@
+import RSVP from 'rsvp';
+import { run } from '@ember/runloop';
+import { DEBUG } from '@glimmer/env';
+import { registerWaiter } from '@ember/test';
+
+const USE_RAF = typeof requestAnimationFrame === 'function';
+let _didTransition;
+let _whenRoutePainted;
+let _whenRouteIdle;
+
+reset();
+
+export const TRANSITION_INTERUPTED = 'TRANSITION INTERUPTED';
+
+export function beginTransition() {
+  _checkForPriorTransition();
+
+  _didTransition = _defer();
+  _whenRoutePainted = _didTransition.promise.then(() => _afterNextPaint());
+  _whenRouteIdle = _whenRoutePainted.then(() => _afterNextPaint());
```

Comment:
I'm a fan. Still unconvinced it is a meaningful queue but at least now folks will have an easier time groking intent.